src/test/java/de/adorsys/sdjwt/sdjwtvp/SdJwtVPTest.java

```diff
@@ -186,4 +189,21 @@ public void testS6_2_PresentationPartialDisclosure() throws ParseException, JOSE
                 .verifySignature(TestSettings.verifierContextFrom(presenteSdJwtVP.getCnfClaim(), "ES256"));
     }
 
-}
+    @Test
+    public void testOf_validInput() {
+    String sdJwtString = TestUtils.readFileAsString(getClass(), "sdjwt/s6.2-presented-sdjwtvp.txt");
+    SdJwtVP sdJwtVP = SdJwtVP.of(sdJwtString);
+
+    assertNotNull(sdJwtVP);
+    assertEquals(4, sdJwtVP.getDisclosures().size());
+    }
+
+    @Test
+    public void testOf_MalformedSdJwt_ThrowsIllegalArgumentException() {
+        // Given
+        String malformedSdJwt = "issuer-signed-jwt"; // missing delimiter at the end
+
+        assertThrows(IllegalArgumentException.class, () -> SdJwtVP.of(malformedSdJwt));
```

Comment:
Consider asserting the exception message.

```suggestion

        var exception = assertThrows(IllegalArgumentException.class, () -> SdJwtVP.of(malformedSdJwt));
        
        assertEquals("SD-JWT is malformed, expected to end with ~", exception.getMessage());
```